crates/astria-sequencer-utils/src/activation_point_estimator.rs

```diff
@@ -0,0 +1,288 @@
+                .await
+                .wrap_err("failed to parse block response as UTF-8 string")
+        };
+
+        let response = tokio::time::timeout(Duration::from_secs(5), blocking_getter)
+            .await
+            .wrap_err_with(|| format!("timed out fetching block at height {height}"))??
+            .trim()
+            .to_string();
+        let json_rpc_response: Value = serde_json::from_str(&response)
+            .wrap_err_with(|| format!("failed to parse block response `{response}` as json"))?;
+        let time_str = json_rpc_response
+            .get("result")
+            .and_then(|value| value.get("block"))
+            .and_then(|value| value.get("header"))
+            .and_then(|value| value.get("time"))
+            .and_then(Value::as_str)
+            .ok_or_else(|| {
+                eyre!(
+                    "expected block response `{response}` to have string field \
+                     `result.block.header.time`"
+                )
+            })?;
+        time_str
+            .parse()
+            .wrap_err_with(|| format!("expected time `{time_str}` to convert to `Timestamp`"))
+    }
+
+    async fn get_network_name(&self) -> Result<String> {
+        let blocking_getter = async {
+            reqwest::get(format!("{}/status", self.sequencer_url))
+                .await
+                .wrap_err("failed to get status")?
+                .text()
+                .await
+                .wrap_err("failed to parse status response as UTF-8 string")
+        };
+
+        let response = tokio::time::timeout(Duration::from_secs(5), blocking_getter)
+            .await
+            .wrap_err("timed out fetching status")??
+            .trim()
+            .to_string();
+        let json_rpc_response: Value = serde_json::from_str(&response)
+            .wrap_err_with(|| format!("failed to parse status response `{response}` as json"))?;
+        Ok(json_rpc_response
+            .get("result")
+            .and_then(|value| value.get("node_info"))
+            .and_then(|value| value.get("network"))
+            .and_then(Value::as_str)
+            .ok_or_else(|| {
+                eyre!(
+                    "expected status response `{response}` to have string field \
+                     `result.node_info.network`"
+                )
+            })?
+            .to_string())
+    }
+}
+
+struct Estimator {
+    current_height: u64,
+    current_timestamp: Timestamp,
+    estimated_block_time_nanoseconds: f64,
+    network_name: String,
+    verbose: bool,
+}
+
+impl Estimator {
+    async fn new(args: &Args) -> Result<Self> {
+        let (current_height, current_timestamp) = args.get_current_height().await?;
+        ensure!(
+            current_height > 1,
+            "need current height to be greater than 1"
+        );
+        let range = std::cmp::min(args.range, current_height);
+        let old_timestamp = args.get_timestamp_at_height(current_height - range).await?;
```

Comment:
If `range = current_height`, then we will `get_timestamp_at_height(0)` which will return an error response.
```suggestion
        let range = std::cmp::min(args.range, current_height.saturating_sub(1));
        let old_timestamp = args.get_timestamp_at_height(current_height - range).await?;
```

Reply:
Good call - thanks! Fixed in [3a14b8a](https://github.com/astriaorg/astria/pull/2114/commits/3a14b8a236dbad758a6b54657bf21b17bbee2dbe).